src/update_agent/actor.rs

```diff
@@ -148,15 +175,30 @@ impl UpdateAgent {
         Box::new(upgrade)
     }
 
+    /// Finalize a deployment (unlock and reboot).
+    fn finalize_deployment(
+        &mut self,
+        can_finalize: bool,
+        release: Release,
+    ) -> ResponseActFuture<Self, Release, ()> {
+        if !can_finalize {
+            return Box::new(actix::fut::err(()));
+        }
+
+        let msg = rpm_ostree::FinalizeDeployment { release };
+        let upgrade = self
+            .rpm_ostree_actor
+            .send(msg)
+            .flatten()
+            .map_err(|e| log::error!("failed to finalize update: {}", e))
```

Comment:
So let's say it fails (either because of [the reboot race](https://github.com/projectatomic/rpm-ostree/issues/1348), in which case it doesn't matter much, OR because it actually couldn't finalize), will the state machine stay in the `UpdateStaged` state and retry on the next tick?

Reply:
Yes, as we would still be returning an error (of type `()`). This is just doing error-type mapping, in order to log error messages without the caller caring about the details. The future chained after this (via a `map()`) will only run on success.

Reply:
Gotcha, thanks! Didn't mean to call out this specific line. Just confirming my understanding at a high level.